### PR TITLE
feat: add Dashboard and DashboardLayout components

### PR DESCRIPTION
**Pull request by tomivirkki**
## Description

Add the `Dashboard` and `DashboardLayout` React wrappers

## Type of change

Feature

### CODE REVIEW

File: packages/react-components-pro/package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "@vaadin/charts": "24.5.0-beta1",
     "@vaadin/cookie-consent": "24.5.0-beta1",
     "@vaadin/crud": "24.5.0-beta1",
+    "@vaadin/dashboard": "24.6.0-alpha0",
```
Comment (web-padawan) [inline packages/react-components-pro/package.json:33]: Let's rebase and update to alpha1.
```suggestion
    "@vaadin/dashboard": "24.6.0-alpha1",
```

Comment (tomivirkki) [reply packages/react-components-pro/package.json]: Ah, still need to update lock file

Comment (tomivirkki) [reply packages/react-components-pro/package.json]: Done

File: test/Grid.spec.tsx [hunk 2]
```diff
@@ -393,8 +393,8 @@ describe('Grid', () => {
         element.dispatchEvent(new MouseEvent('dblclick', { bubbles: true }));
       }
 
-      function focusOut(element: Element) {
-        element.dispatchEvent(new FocusEvent('focusout', { bubbles: true }));
+      function focusOut(element: HTMLElement) {
+        element.blur();
```
Comment (web-padawan) [inline test/Grid.spec.tsx:397]: Could maybe fix grid-pro tests in a separate PR and backport to v24.5 in case it's also broken there.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**